pkg/storaged/btrfs/volume.jsx

```diff
@@ -0,0 +1,190 @@
+    // only passes the device and not the mountpoint when the device is mounted.
+    // https://github.com/storaged-project/libblockdev/issues/966
+    const is_mounted = btrfs_is_volume_mounted(client, block_devices);
+
+    function rename_dialog() {
+        dialog_open({
+            Title: _("Change label"),
+            Fields: [
+                TextInput("name", _("Name"),
+                          {
+                              validate: name => validate_fsys_label(name, "btrfs"),
+                              value: label
+                          })
+            ],
+            Action: {
+                Title: _("Save"),
+                action: function (vals) {
+                    return block_btrfs.SetLabel(vals.name, {});
+                }
+            }
+        });
+    }
+
+    return (
+        <Stack hasGutter>
+            <StackItem>
+                <StorageCard card={card}>
+                    <CardBody>
+                        <DescriptionList className="pf-m-horizontal-on-sm">
+                            <StorageDescription title={_("Label")}
+                                                value={label}
+                                                action={
+                                                    <StorageLink onClick={rename_dialog}
+                                                               excuse={is_mounted ? _("Btrfs volume is mounted") : null}>
+                                                        {_("edit")}
+                                                    </StorageLink>}
+                            />
+                            <StorageDescription title={_("UUID")} value={uuid} />
+                            <StorageDescription title={_("Capacity")} value={fmt_size_long(use[1])} />
+                            <StorageDescription title={_("Usage")}>
+                                <StorageUsageBar key="s" stats={use} />
+                            </StorageDescription>
+                        </DescriptionList>
+                    </CardBody>
+                </StorageCard>
+            </StackItem>
+        </Stack>
+    );
+};
+
+const BtrfsDevicesCard = ({ card, uuid }) => {
+    return (
+        <StorageCard card={card}>
+            <CardBody className="contains-list">
+                <PageTable emptyCaption={_("No devices found")}
+                           aria-label={_("btrfs device")}
+                           crossrefs={get_crossrefs(uuid)} />
+            </CardBody>
+        </StorageCard>);
+};
+
+const BtrfsSubVolumesCard = ({ card, volume }) => {
+    return (
+        <StorageCard card={card}>
+            <CardBody className="contains-list">
+                <ChildrenTable emptyCaption={_("No subvolumes")}
+                               aria-label={_("btrfs subvolumes")}
+                               page={card.page} />
+            </CardBody>
+        </StorageCard>
```

Comment:
These 9 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19690-20231205-110900-7b3be2c9-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)